concepts/modularization/README.md

```diff
@@ -0,0 +1,75 @@
+# Modularization
+
+Kyma provides Kubernetes building blocks. It should be easy to pick only those that are needed for the job and it should be easy to add new blocks to extend Kyma features. With the growing number of components, it is not possible to always install them all anymore. 
+
+- [Independent releases of modules](#independent-releases-of-modules)
+- [Dependencies between components](#dependencies-between-components)
+- [Release channels](#release-channels)
+- [Versioning](#versioning)
+- [Manifest operator](#manifest-operator)
+
+# Independent releases of modules
+With the growing number of components, it is hard to deliver features and fixes quickly and efficiently. Changes in manifests require a new release of Kyma. Operators (reconcilers) are tightly coupled and must be released together. In most cases, new component releases don't involve any API changes and could be delivered in a few minutes. 
+
+
+# Dependencies between components
+Components can depend only on core Kubernetes API or API extensions introduced by other components. Component operators are responsible for checking if the required APIs are available and react properly to the missing dependencies by reducing functionality or even reporting errors. Component owners are responsible for integration tests with their dependencies (with all versions supported in official release channels).
```

Comment:
if a component requires another component to work correctly, what does `react properly` mean? If it just reports an error who would resolve this error? Is it the users job to then install the required component as well?

Reply:
If the API you require is not available you should fail (e.g. core kubernetes API, or istio virtual service). If your component can work without the API, but some features are not available (e.g. service monitor from monitoring) you should just skip it. We can discuss if it is useful to report such partial degradation in the status (and how).